animated_sprite.cpp

```diff
@@ -0,0 +1,113 @@
+#include "animated_sprite.h"
+#include <iostream>
+
+// AnimatedSprite()
+AnimatedSprite::AnimatedSprite(std::string sprite_sheet_name,
+                               SDL_Rect first_sprite_coord_location,
+                               SDL_Renderer* renderer,
+                               float time_to_update_sprite,
+                               SDL_Rect destination)
+    : Sprite(sprite_sheet_name, first_sprite_coord_location, renderer),
+      time_to_update(time_to_update_sprite),
+      current_animation_once(false),
+      current_animation(""),
+      frame_index_(0),
+      visible_(true),
+      destination_(destination) {
+  this->SetupAnimations();
+}
+
+// TimeToUpdateFrame()
+void AnimatedSprite::TimeToUpdateFrame() {
+  if (elapsed_time_ >= time_to_update) {
+    if (frame_index_ < (int)animations_[this->current_animation].size() - 1) {
+      frame_index_++;
+    } else {
+      if (current_animation_once == true) {
+        this->SetVisible(false);
+      }
+      frame_index_ = 0;
+    }
+    elapsed_time_ = 0;
+  }
+}
+
+// GetCurrentAnimation()
+std::string AnimatedSprite::GetCurrentAnimation() { return current_animation; }
+
+// GetDestinationRect()
+SDL_Rect AnimatedSprite::GetDestinationRect() { return destination_; }
+
+// GetFrameIndex()
+int AnimatedSprite::GetFrameIndex() { return frame_index_; }
+
+// GetOffsets()
+Vector2 AnimatedSprite::GetOffsets(std::string animation) {
+  return offsets_[animation];
+}
+
+// GetSourceRect()
+SDL_Rect AnimatedSprite::GetSourceRect(std::string animation, int frame) {
+  return animations_[animation][frame];
+}
+
+// GetVisible()
+bool AnimatedSprite::GetVisible() { return visible_; }
+
+// SetCurrentAnimation()
+void AnimatedSprite::SetCurrentAnimation(std::string name) {
+  current_animation = name;
+}
+
+// SetCurrentAnimationOnce()
+void AnimatedSprite::SetCurrentAnimationOnce(bool once) {
+  current_animation_once = once;
+}
+
+// SetDestinationRect()
+void AnimatedSprite::SetDestinationRect(SDL_Rect new_rect) {
```

Comment:
Take these parameters by const ref unless you can't. Passing by value makes a copy.

```
foo(const SDL_Rect& new_destination) {}
```

Reply:
fix